topi/include/topi/nn/pooling.h

```diff
@@ -129,19 +129,26 @@ inline Tensor pool_impl(const Tensor& x,
       return tvm::max(temp(indices), { dheight, dwidth });
     }, "tensor", "pool_max");
   } else if (pool_type == kAvgPool) {
+    // Pad the inputs
     auto temp = do_pad ? pad(x, pad_before, pad_after, 0, "pad_temp") : x;
-    auto tavg = [&](const Array<Var>& output, Expr divide_factor) {
+
+    // TVM compute for summing the pooling window.
+    auto pool_sum = tvm::compute(out_shape,
+    [&](const Array<Var>& output) {
       Array<Expr> indices;
       for (const Var& var : output) indices.push_back(var);
       indices.Set(height_axis, output[height_axis] * stride_height + dheight);
       indices.Set(width_axis, output[width_axis] * stride_width + dwidth);
-      return tvm::sum(temp(indices) / divide_factor, { dheight, dwidth });
-    };
+      return tvm::sum(temp(indices), { dheight, dwidth });
```

Comment:
If we could leave one comment of TODO, it is better. Remind us there is one potential issue we should care in the future parsing TFLite model.

Reply:
@FrozenGene If you mean int8<->uint8 casting for tflite, I think the TODO can be put into tflite frontend instead. The bugfix here in topi is complete IMO. what do you think?

Reply:
I mean the `tvm.sum` type is `uint16`, not `int32`.

Reply:
For TFLite adaptation, we can `cast` the input to `uint16` and then pass the casted input to `avg_pool2d`. This is purely the job of framework parser. It does not need any code in the TOPI layer.

Reply:
yes my understanding is `avg_pool2d(int_8_value.astype("uint16")).astype("int8")` will be in frontend, right?

Reply:
I understand the way you want to do. Even the output dtype is "int32", you also want to do `avg_pool2d(int_8_value.astype("int32")).astype("int8")` in frontend.

Reply:
If so, the way is ok.